### PR TITLE
DYN-6151 Add dash to note crashing Dynamo

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Per user crash report at https://cer.autodesk.com/search/reports/714240100/, add defensive code to guard against potential crash on note edition.

- [x] Add try catch in string utility function
- [x] Add try catch in note key down handler because according to the CER report, the actual crash happened at the textbox select command, which is not common. This could be a threading/ memory issue not related to Dynamo.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

add defensive code to guard against potential crash on note edition

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Engine/ProtoCore/Utils/StringUtils.cs [hunk 1]
```diff
@@ -66,14 +66,40 @@ public static string[] BreakTextIntoLines(string text)
...
+        /// </summary>
+        /// <param name="text"></param>
+        /// <param name="tabSpacingSize"></param>
+        /// <returns></returns>
         public static string SpaceToTabConversion(string text, int tabSpacingSize)
         {
-            return text.Replace(new String(' ', tabSpacingSize), "\t");
+            try
+            {
+                return text.Replace(new string(' ', tabSpacingSize), "\t");
```
Comment (zeusongit) [inline src/Engine/ProtoCore/Utils/StringUtils.cs:97]: typo here: 's' should be capitalized to 'S'

**Review by zeusongit — APPROVED** [review]
LGTM, one change requested.

File: src/Engine/ProtoCore/Utils/StringUtils.cs (hunk 1 above)
Comment (QilongTang) [reply src/Engine/ProtoCore/Utils/StringUtils.cs:97]: Hi @zeusongit This is actually a change suggested by Visual Studio

Comment (zeusongit) [reply src/Engine/ProtoCore/Utils/StringUtils.cs:97]: Oh, all good in that case.

File: src/DynamoCoreWpf/Views/Core/NoteView.xaml.cs [hunk 2]
```diff
@@ -19,6 +19,12 @@ namespace Dynamo.Nodes
     public partial class NoteView : IViewModelView<NoteViewModel>
     {
         private EditWindow editWindow;
+
+        /// <summary>
+        /// Special keys defination in note
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Core/NoteView.xaml.cs:24]: sp(definition)

**Pull request merged**